src/psyclone/tests/psyir/nodes/reference_test.py

```diff
@@ -94,47 +109,60 @@ def test_reference_symbol(monkeypatch):
     # Symbol in KernelSchedule SymbolTable
     field_old = references[0]
     assert field_old.name == "field_old"
-    assert isinstance(field_old.symbol(), DataSymbol)
-    assert field_old.symbol().name == field_old.name
+    assert isinstance(field_old.symbol, DataSymbol)
+    assert field_old.symbol in kernel_schedule.symbol_table.symbols
 
     # Symbol in KernelSchedule SymbolTable with KernelSchedule scope
-    assert isinstance(field_old.symbol(scope_limit=kernel_schedule),
+    assert isinstance(Reference.get_symbol(field_old.name, field_old,
+                                           scope_limit=kernel_schedule),
                       DataSymbol)
-    assert field_old.symbol().name == field_old.name
+    assert field_old.symbol.name == field_old.name
 
     # Symbol in KernelSchedule SymbolTable with parent scope
-    assert field_old.symbol(scope_limit=field_old.parent) is None
+    with pytest.raises(SymbolError) as excinfo:
```

Comment:
Please update the comment above, e.g. "... SymbolTable should not be found if we limit the scope to the immediate parent of the reference" or somesuch.

Reply:
Added.